### PR TITLE
Feature/no more vulns

### PR DESCRIPTION
**Pull request by pierreTklein**
### Tickets:

[HCK-178](https://hackmcgill.atlassian.net/browse/HCK-178?atlOrigin=eyJpIjoiMzYwNTFlYWQzZGRmNGMxYTg1NWE4ODFkNzRjNjBhM2YiLCJwIjoiaiJ9)
Screen Shot 2019-12-14 at 1 50 31 PM

### List of changes:

- `npm run update`, `npm audit fix --force`
- Replace `location.X` with `window.location.X`
- Remove empty constructors 

### Type of change:

- [X] Bug fix (non-breaking change which fixes an issue)

### How did you do this?

Running npm script and seeing what errors / warnings pop up

### Why are you choosing this approach?

It's the only way to do it.

### Questions for code reviewers?

Verify no regression.

### PR Checklist:

- [X] Merged `develop` branch (before testing)
- [X] Linted my code locally
- [x] Listed change(s) in the Changelog *(Where is this changelog)*
- [X] Tested all links in project relevant browsers
- [X] Tested all links on different screen sizes
- [X] Referenced all useful info (issues, tasks, etc)

### CODE REVIEW

**Comment by pierreTklein** [comment]
> changelog is in `/docs/CHANGELOG.md`

Done.

File: src/features/404/404.tsx [hunk 1]
```diff
@@ -16,9 +16,6 @@ import Paragraph from '../../shared/Elements/Paragraph';
  * Container that renders 404 not found page.
  */
 class NotFoundContainer extends React.Component<RouteComponentProps> {
-  constructor(props: RouteComponentProps) {
-    super(props);
-  }
   public render() {
```
Comment (RohitGarudadri) [inline src/features/404/404.tsx]: Removing this why?

File: src/features/Checkin/Main.tsx [hunk 2]
```diff
@@ -139,7 +139,7 @@ class CheckinContainer extends React.Component<{}, ICheckinState> {
         ':id',
         '[a-f\\d]{24}'
       );
-      const hackerURL = new RegExp(`^http(s)?:\/\/.+${subRouteRegex}`);
```
Comment (RohitGarudadri) [inline src/features/Checkin/Main.tsx]: Does it still work without the backslashes?

File: src/features/Login/PasswordForgotConfirmation.tsx [hunk 3]
```diff
@@ -15,9 +15,6 @@ import {
 } from '../../shared/Elements';
 
 class PasswordResetEmailConfirmationContainer extends React.Component<{}, {}> {
-  constructor(props: {}) {
-    super(props);
-  }
```
Comment (RohitGarudadri) [inline src/features/Login/PasswordForgotConfirmation.tsx]: DItto

File: src/shared/HOC/withNavbar.tsx [hunk 4]
```diff
@@ -3,10 +3,6 @@ import Navbar from '../../features/Nav/Navbar';
 
 const withNavbar = <P extends {}>(Component: React.ComponentType<P>) =>
   class extends React.Component<P> {
-    constructor(props: any) {
-      super(props);
-    }
-
```
Comment (RohitGarudadri) [inline src/shared/HOC/withNavbar.tsx]: DItto

File: src/shared/HOC/withThemeProvider.tsx [hunk 5]
```diff
@@ -5,10 +5,6 @@ import theme from '../Styles/theme';
 
 const withThemeProvider = <P extends {}>(Component: React.ComponentType<P>) =>
   class extends React.Component<P> {
-    constructor(props: any) {
-      super(props);
-    }
-
```
Comment (RohitGarudadri) [inline src/shared/HOC/withThemeProvider.tsx]: DItto

File: src/shared/HOC/withToaster.tsx [hunk 6]
```diff
@@ -3,10 +3,6 @@ import { ToastContainer } from 'react-toastify';
 
 const WithToaster = <P extends {}>(Component: React.ComponentType<P>) =>
   class extends React.Component<P> {
-    constructor(props: any) {
-      super(props);
-    }
-
```
Comment (RohitGarudadri) [inline src/shared/HOC/withToaster.tsx]: DItto

**Comment by pierreTklein** [comment]
all code changes are due to generated warnings from TSLint. One warning was for empty constructors. Another was for redundant use of \, since the regexp is surrounded by double quotes.

File: src/shared/HOC/withToaster.tsx (hunk 6 above)
Comment (pierreTklein) [reply src/shared/HOC/withToaster.tsx]: Check comment below

**Pull request merged**